src/App.tsx

```diff
@@ -1,11 +1,26 @@
 import { useState } from 'react';
+
 import './App.scss';
+
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
+
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
```

Comment:
You could hold in state a map of movies (an object with id's as keys) eg.:
```js
{
  '1':  {
    title: 'tytul',
    description: 'plot',
    imdbId: 'id',
    imgUrl: 'poster',
    imdbUrl: `url`,
  }
}
```
Then you wouldn't have to iterate of an array in addMovie everytime you want to add a new one and you could pass to the FindMovie only setMovies

Reply:
Or still only pass the setMovies to the FindMovie component and when setting the state do the filtering inside:
```js
setMovies(prevState => {
  if (!prevState.some(prevMovie => prevMovie.imdbId === movie.imdbId)) {
    return [...prevState, movie];
  }

  return prevState;
});
```

or

```js
setMovies(prevState => [...prevState, ...(prevState.some(prevMovie => prevMovie.imdbId === movie.imdbId) ? [] : [movie])])
```
no need for a separate function :)